style: apply /style-guide pass to models/registry (#2721)
wandb/docs: models/registry/aliases.mdx

description: "Use default, custom, and protected aliases to reference specific artifact versions in W&B Registry."
---

> Aliases give you a stable, human-readable way to point to a specific [artifact version](/models/artifacts/create-a-new-artifact-version/) in W&B Registry, so you can track which version is current, in production, or otherwise notable without having to remember version numbers. [W&B automatically assigns aliases](/models/registry/aliases/#default-aliases) to each artifact you link with the same name, and you can also [create one or more custom aliases](/models/registry/aliases/#custom-aliases) to reference a specific artifact version. This page explains the default aliases that W&B assigns automatically, how to create your own custom aliases, and how to use protected aliases to safeguard important versions.

Reviewer: Too markety, wordy

```suggestion
Use aliases to reference a specific [artifact version](/models/artifacts/create-a-new-artifact-version/) in W&B Registry. Aliases provide stable, human-readable labels for versions that are current, in production, or otherwise important. [W&B automatically assigns aliases](/models/registry/aliases/#default-aliases) to each artifact you link with the same name. You can also [create custom aliases](/models/registry/aliases/#custom-aliases) and use protected aliases to safeguard important versions.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

## Default aliases

> W&B automatically assigns the following aliases to each artifact version you link with the same name, so you can identify a version without configuring anything:

Reviewer: ```suggestion
W&B automatically assigns the following aliases to each artifact version you link with the same name:
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

## Custom aliases

> Custom aliases let you label artifact versions with names that reflect your workflow, such as the dataset a model was trained on or which version is performing best. For example:

Reviewer: ```suggestion
Use custom aliases to label artifact versions by workflow, dataset, performance, or other criteria. For example:
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

## Custom aliases

Custom aliases let you label artifact versions with names that reflect your workflow, such as the dataset a model was trained on or which version is performing best. For example:

- You might use aliases such as `dataset_version_v0`, `dataset_version_v1`, and `dataset_version_v2` to identify which dataset a model was trained on.
- You might use a `best_model` alias to keep track of the best performing artifact model version.

> Any user with a [**Member** or **Admin** registry role](/models/registry/configure_registry/#registry-roles) on a registry can add or remove a custom alias from a linked artifact in that registry. Users with the [**Restricted Viewer** or **Viewer** roles](/models/registry/configure_registry/#registry-roles) can't add or remove aliases.

Author: Fixed in f1f7b2c — both links now point to `#role-permissions`, the table that defines each role's capabilities.

Note: `configure_registry.mdx` never had a literal `#registry-roles` heading — the sections are "Configure registry roles" (`#configure-registry-roles`) and "Details about registry roles" (`#details-about-registry-roles`) — so these links were already broken on `main`.

4. Click the **+** button to add one or more aliases next to the **Aliases** field.
</Tab>
<Tab title="Python SDK">
When you link an artifact version to a collection with the Python SDK, you can optionally provide a list of one or more aliases as an argument to the `aliases` parameter in [`link_artifact()`](/models/ref/python/experiments/run/#method-runlink_artifact). W&B creates an alias ([non protected alias](#custom-aliases)) for you if the alias you provide doesn't already exist.

> The following code snippet demonstrates how to link an artifact version to a collection and add aliases to that artifact version with the Python SDK. Replace the bracketed placeholders with your own values:

Reviewer: ```suggestion
The following code snippet demonstrates how to link an artifact version to a collection and add aliases to that artifact version with the Python SDK. Replace values enclosed in `<>` with your own values:
```

Author: Applied your suggestion; the code sample now uses `<>` placeholders to match.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

```python
import wandb

# Initialize a run
with wandb.init(entity = "[TEAM-ENTITY]", project = "[PROJECT-NAME]") as run:

    # Create an artifact object
    # The type parameter specifies both the type of the 
    # artifact object and the collection type
    artifact = wandb.Artifact(name = "[NAME]", type = "[TYPE]")

    # Add the file to the artifact object. 
    # Specify the path to the file on your local machine.
    artifact.add_file(local_path = "[LOCAL-PATH-TO-ARTIFACT]")

    # Specify the collection and registry to link the artifact to
    REGISTRY_NAME = "[REGISTRY-NAME]"
    COLLECTION_NAME = "[COLLECTION-NAME]"
    target_path=f"wandb-registry-{REGISTRY_NAME}/{COLLECTION_NAME}"

    # Link the artifact version to the collection
    # Add one or more aliases to this artifact version
    run.link_artifact(
        artifact = artifact, 
        target_path = target_path, 

>         aliases = ["[ALIAS-1]", "[ALIAS-2]"]

Reviewer: Don't use square brackets. Makes it looks like lists.

Author: Changed these to `<alias-1>`/`<alias-2>`, and converted the rest of the sample to `<lowercase>` angle brackets.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

        )
```
</Tab>
</Tabs>

### Protected aliases

Protected aliases combine labeling with a guardrail: besides identifying a version, they prevent that version from being modified or deleted, which is useful for releases that downstream systems depend on. For example, consider a `production` protected alias to label and identify artifact versions that are in use in your organization's machine learning production pipeline.

> [Registry admin](/models/registry/configure_registry/#role-permissions) users and [service accounts](/support/models/articles/what-is-a-service-account-and-why-is-it-) with the **Admin** role can create protected aliases and add or remove protected aliases from an artifact version. Users and service accounts with **Member**, **Viewer**, and **Restricted Viewer** roles can't unlink a protected version or delete a collection that contains a protected alias. See [Configure registry access](/models/registry/configure_registry/) for details.

Reviewer: ```suggestion
Use [protected alias](/models/registry/aliases/#protected-aliases) to identify an artifact version and prevent it from being modified or deleted. Protected aliases are useful for versions that downstream systems depend on. For example, use a `production` protected alias for artifact versions used in your organization's production ML pipeline.

[Registry admin](/models/registry/configure_registry/#role-permissions) users and [service accounts](/support/models/articles/what-is-a-service-account-and-why-is-it-) with the **Admin** role can create protected aliases and add or remove protected aliases from an artifact version. Users and service accounts with **Member**, **Viewer**, and **Restricted Viewer** roles can't unlink a protected version or delete a collection that contains a protected alias. See [Configure registry access](/models/registry/configure_registry/) for details.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

4. Click the **+** button to add one or more aliases next to the **Aliases** field.
</Tab>
<Tab title="Python SDK">

> When you link an artifact version to a collection with the Python SDK, you can optionally provide a list of one or more aliases as an argument to the `alias` parameter in [`link_artifact()`](/models/ref/python/experiments/run.md/#link_artifact). W&B creates an alias ([non protected alias](#custom-aliases)) for you if the alias you provide doesn't already exist.

Author: Fixed in f1f7b2c. Corrected the parameter name to `aliases` (matching the code sample below) and the reference link to `/models/ref/python/experiments/run/#method-runlink_artifact`, which matches the `### method Run.link_artifact` anchor in `run.mdx` and the convention used elsewhere (e.g. `#method-runuse_artifact`).

    # artifact object and the collection type
    artifact = wandb.Artifact(name = "[NAME]", type = "[TYPE]")

    # Add the file to the artifact object. 
    # Specify the path to the file on your local machine.
    artifact.add_file(local_path = "[LOCAL-PATH-TO-ARTIFACT]")

    # Specify the collection and registry to link the artifact to
    REGISTRY_NAME = "[REGISTRY-NAME]"
    COLLECTION_NAME = "[COLLECTION-NAME]"
    target_path=f"wandb-registry-{REGISTRY_NAME}/{COLLECTION_NAME}"

    # Link the artifact version to the collection
    # Add one or more aliases to this artifact version
    run.link_artifact(
        artifact = artifact, 
        target_path = target_path, 
        aliases = ["[ALIAS-1]", "[ALIAS-2]"]
        )
```
</Tab>
</Tabs>

### Protected aliases

Protected aliases combine labeling with a guardrail: besides identifying a version, they prevent that version from being modified or deleted, which is useful for releases that downstream systems depend on. For example, consider a `production` protected alias to label and identify artifact versions that are in use in your organization's machine learning production pipeline.

> [Registry admin](/models/registry/configure_registry/#registry-roles) users and [service accounts](/support/models/articles/what-is-a-service-account-and-why-is-it-) with the **Admin** role can create protected aliases and add or remove protected aliases from an artifact version. Users and service accounts with **Member**, **Viewer**, and **Restricted Viewer** roles can't unlink a protected version or delete a collection that contains a protected alias. See [Configure registry access](/models/registry/configure_registry/) for details.

Author: Fixed in f1f7b2c — repointed to `#role-permissions`, same as the line 37 links.

    # Add the file to the artifact object. 
    # Specify the path to the file on your local machine.
    artifact.add_file(local_path = "<local-path-to-artifact>")

    # Specify the collection and registry to link the artifact to
    REGISTRY_NAME = "<registry-name>"
    COLLECTION_NAME = "<collection-name>"
    target_path=f"wandb-registry-{REGISTRY_NAME}/{COLLECTION_NAME}"

    # Link the artifact version to the collection
    # Add one or more aliases to this artifact version
    run.link_artifact(
        artifact = artifact, 
        target_path = target_path, 
        aliases = ["<alias-1>", "<alias-2>"]
        )
```
</Tab>
</Tabs>

### Protected aliases

> Use [protected alias](/models/registry/aliases/#protected-aliases) to identify an artifact version and prevent it from being modified or deleted. Protected aliases are useful for versions that downstream systems depend on. For example, use a `production` protected alias for artifact versions used in your organization's production ML pipeline.

Reviewer 2: ```suggestion
Use a [protected alias](/models/registry/aliases/#protected-aliases) to identify an artifact version and prevent it from being modified or deleted. Protected aliases are useful for versions that downstream systems depend on. For example, use a `production` protected alias for artifact versions used in your organization's production ML pipeline.
```

Author: Added the article — now "Use a [protected alias]…".

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

After creation, each protected alias appears as a gray rectangle with a lock icon in the **Protected Aliases** section.  

<Note>

> Unlike custom aliases that aren't protected, you can create protected aliases only in the W&B Registry UI, not programmatically with the Python SDK. To add a protected alias to an artifact version, you can use the W&B Registry UI or the Python SDK.

Reviewer: Check if we refer to W&B Registry UI or W&B Registry or W&B Registry App (ask Matt)

Reviewer 2: I think John has been changing "W&B App" to "W&B UI" to avoid confusion with the iOS app. I'm fine with "W&B Registry UI" here, if I'm correct about the trajectory.

Author: Keeping "W&B Registry UI" here per @mdlinville's confirmation below — this resolves the "ask Matt" question.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

3. Within the **Versions** section, select the **View** button for a specific artifact version.
4. Click the **+** button to add one or more protected aliases next to the **Aliases** field.

> After an admin creates a protected alias, the admin can add it to an artifact version programmatically with the Python SDK. For an example of how to add a protected alias to an artifact version, see [Custom aliases](#custom-aliases).

Reviewer 2: ```suggestion
After an admin creates a protected alias, it can be added to an artifact version either in the UI or programmatically with the Python SDK. See [Custom aliases](#custom-aliases).
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

## Find existing aliases

> Once you have aliases in place, you can find them with the [global search bar in the W&B Registry](/models/registry/search_registry/#search-for-registry-items). To find a protected alias:

Reviewer 2: ```suggestion
You can find aliases with the [global search bar in the W&B Registry](/models/registry/search_registry/#search-for-registry-items). To find a protected alias:
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
